sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayServiceConfigurationReader.java

```diff
@@ -41,118 +41,32 @@
  */
 public class GatewayServiceConfigurationReader {
 
-    public static final String GATEWAY_READER_NOT_INITIALIZED = "GatewayServiceConfigurationReader has not been initialized";
-
-    public ReplicationPolicy userReplicationPolicy;
-    private ReplicationPolicy systemReplicationPolicy;
-    private ConsistencyLevel consistencyLevel;
-    private volatile boolean initialized;
     private URI serviceEndpoint;
-    private final ConnectionPolicy connectionPolicy;
-    private Map<String, Object> queryEngineConfiguration;
-    private final BaseAuthorizationTokenProvider baseAuthorizationTokenProvider;
-    private final boolean hasAuthKeyResourceToken;
-    private final String authKeyResourceToken;
-    private HttpClient httpClient;
+    private GlobalEndpointManager globalEndpointManager;
 
-    public GatewayServiceConfigurationReader(URI serviceEndpoint, boolean hasResourceToken, String resourceToken,
-            ConnectionPolicy connectionPolicy, BaseAuthorizationTokenProvider baseAuthorizationTokenProvider,
-            HttpClient httpClient) {
+    public GatewayServiceConfigurationReader(URI serviceEndpoint, GlobalEndpointManager globalEndpointManager) {
         this.serviceEndpoint = serviceEndpoint;
-        this.baseAuthorizationTokenProvider = baseAuthorizationTokenProvider;
-        this.hasAuthKeyResourceToken = hasResourceToken;
-        this.authKeyResourceToken = resourceToken;
-        this.connectionPolicy = connectionPolicy;
-        this.httpClient = httpClient;
+        this.globalEndpointManager = globalEndpointManager;
+        this.globalEndpointManager.getDatabaseAccountFromCache(this.serviceEndpoint).block();
     }
 
     public ReplicationPolicy getUserReplicationPolicy() {
-        this.throwIfNotInitialized();
-        return this.userReplicationPolicy;
+        DatabaseAccount databaseAccount = this.globalEndpointManager.getDatabaseAccountFromCache(this.serviceEndpoint).block();
+        return BridgeInternal.getReplicationPolicy(databaseAccount);
     }
 
     public ReplicationPolicy getSystemReplicationPolicy() {
-        this.throwIfNotInitialized();
-        return this.systemReplicationPolicy;
-    }
-
-    public boolean enableAuthorization() {
-        return true;
+        DatabaseAccount databaseAccount = this.globalEndpointManager.getDatabaseAccountFromCache(this.serviceEndpoint).block();
```

Comment:
This is blocking call on cache , which should be fine as per me. But if anyone have any strong opinion for these methods to return Mono and used asynchronously please comment. I avoided it as it needed good amount of code touch in different layer of SDK where these properties are getting used.

Reply:
blocking calls inside the SDK are not good. they can cause deadlock, perf hit, etc. We are not doing blocking calls anywhere else.

Reply:
We cant really say  at least  for this scenario that this will do deadlock as all are read calls  , even for perf this is cache and not network .But i got your point of avoiding them in code everywhere  and follow async behavior , I will move them to non blocking

Reply:
done